src/picongpu/include/particles/ionization/byCollision/ThomasFermi/ThomasFermi_Impl.hpp

```diff
@@ -272,19 +272,33 @@ namespace ionization
                 /* energy density in sim units */
                 float_X kinEnergyDensity = kinEnergyV[0];
 
-                /* this is the point where actual ionization takes place */
+                /* Thomas-Fermi calculation step:
+                 * Determines the new average charge state for each ion under
+                 * LTE conditions. Returns the new number of bound electrons
+                 * according to this average charge state.
+                 */
                 IonizationAlgorithm ionizeAlgo;
                 float_X newBoundElectrons = ionizeAlgo(
                     kinEnergyDensity, density,
                     particle, this->randomGen()
                     );
 
-                particle[boundElectrons_] = newBoundElectrons;
-
-                /* safety check to avoid double counting since recombination is not yet implemented */
+                /* Safety Check: Would the number of bound electrons after the
```

Comment:
Yeah, I get your point.

My line of thinking was:
Our model does not know anything about implementation. We ask from it what the average charge state depending on local parameters at that very specific point in time would be and have it transfer that answer into number of bound electrons. Then at the place where we decide what we actually do with the ion we find that we only want to change the value `boundElectrons` for the ions in case of ionization. That requires a check, comparing bound electron number before and after. Only if the model predicts that it is lower afterwards we *actually* change it.

Thus I wanted to separate the model and its result from the decision what to do with the result.

I will check this again for consistency. Thanks for the feedback!

@psychocoderHPC suggested to include the option to create electrons with higher weighting instead of multiple macroparticles in case the number of electrons created in one time step is larger than 1.